experimental/cuda2/status_util.c

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "experimental/cuda2/status_util.h"
+
+#include <stddef.h>
+
+#include "experimental/cuda2/dynamic_symbols.h"
+#include "iree/base/status.h"
+
+//===----------------------------------------------------------------------===//
+// CUDA result utilities
+//===----------------------------------------------------------------------===//
+
+// The list of CUDA error strings with their corresponding IREE error state
+// classification.
+//
+// Note that the list of errors is taken from `cudaError_enum` in cuda.h.
+// This is not an exhaustive list; we are just listing common ones here.
+#define CUDA_ERROR_LIST(CUDA_MAP_ERROR)                                        \
```

Comment:
would be nice to IREE_ prefix all of these macros to avoid conflicts/confusion as to where they come from